qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +99,343 @@ class Command(BaseModel):
+            schedule: Union[Schedule, ParameterizedSchedule]) -> None:
+        """
+        Add a new known operation.
+
+        Args:
+            operation: The name of the operation to add.
+            qubits: The qubits which the operation applies to.
+            schedule: The Schedule that implements the given operation.
+
+        Returns:
+            None
+
+        Raises:
+            PulseError: If the qubits are provided as an empty iterable.
+        """
+        qubits = _to_tuple(qubits)
+        if qubits == ():
+            raise PulseError("Cannot add definition {} with no target qubits.".format(operation))
+        if not isinstance(schedule, (Schedule, ParameterizedSchedule)):
+            raise PulseError("Attemping to add an invalid schedule type.")
+        if self.has(operation, qubits):
+            warnings.warn("Replacing previous definition of {} on qubit{} "
+                          "{}.".format(operation,
+                                       's' if len(qubits) > 1 else '',
+                                       qubits if len(qubits) > 1 else qubits[0]))
+        self._ops_def[operation][qubits] = schedule
+
+    def remove(self, operation: str, qubits: Union[int, Iterable[int]]) -> None:
+        """Remove the given operation from the defined operations.
+
+        Args:
+            operation: The name of the operation to add.
+            qubits: The qubits which the operation applies to.
+
+        Returns:
+            None
+
+        Raises:
+            PulseError: If the operation is not present.
+        """
+        self.assert_has(operation, qubits)
+        self._ops_def[operation].pop(_to_tuple(qubits))
+
+    def pop(self,
+            operation: str,
+            qubits: Union[int, Iterable[int]],
+            *params: List[Union[int, float, complex]],
+            **kwparams: Dict[str, Union[int, float, complex]]) -> Schedule:
+        """
+        Remove and return the defined Schedule for the given operation on the given qubits.
+
+        Args:
+            operation: Name of the operation.
+            qubits: The qubits for the operation.
+            *params: Command parameters for generating the output schedule.
+            **kwparams: Keyworded command parameters for generating the schedule.
+
+        Returns:
+            The Schedule defined for the input.
+
+        Raises:
+            PulseError: If command for qubits is not available
+        """
+        self.assert_has(operation, qubits)
+        schedule = self._ops_def[operation].pop(_to_tuple(qubits))
+        if isinstance(schedule, ParameterizedSchedule):
+            return schedule.bind_parameters(*params, **kwparams)
+        return schedule
+
+    def __str__(self):
+        return ('{name}(operations=[{ops}], n_qubits={n_qubits})'.format(
+            name=self.__class__.__name__,
+            ops=self.ops(),
+            n_qubits=len(self._qubit_freq_est_hz)))
+
+    def __repr__(self):
```

Comment:
I suggest not messing around with the `__repr__` for gaining a bit of convenience. That's what `__str__` is for. Otherwise this is how every object of this class will be displayed now, which can be sometimes odd:

<img width="922" alt="image" src="https://user-images.githubusercontent.com/8622381/68562267-e4a48400-0416-11ea-961f-f87f64431b26.png">

Reply:
The current `__str__` is actually not very informative. It's very close to basically saying what the device `basis_gates` are, which you can get from `configuration` anyway. I know this is implementing what was requested in #2861 but the spirit of that issue is to provide a more compact representation of the cmd_def, not to cut it down to just saying what gates are supported.

So I suggest replacing it with the current `__repr__` and avoid overwriting `__repr__`.